### PR TITLE
allow cancellation of call using dio cancellation

### PR DESCRIPTION
**Pull request by rgplvr**
allow cancellation of call using dio cancellation

### CODE REVIEW

**Comment by knaeckeKami** [comment]
Hello, thanks for the PR. It seems there is an issue with the CI, looks like a breaking change in one of the dependencies of multipack. I'll try to fix it.

Support for cancellation in Dio is good, but is passing a signal cancel token for all requests of the link the right approach? 

You can only cancel the token once, right?

I think a better solution would be to pass the cancel token via the context. Then the client code could either create a new cancel token for each request or re-use the same token for multiple requests if preferred.


See e.g. https://github.com/gql-dart/gql/blob/master/links/gql_dio_link/lib/src/dio_link.dart#L312


```dart

class DioLinkCancelTokenContextEntry extends ContextEntry {

     final CancelToken token;

     ...
}
```
in DioLink:


```dart

...
final cancelToken = request.context.entry<DioLinkCancelTokenContextEntry>();

...

```

**Comment by rgplvr** [comment]
Makes sense. I have made the changes to push the cancellation token to the context entry.

**Comment by knaeckeKami** [comment]
please update your branch from master to fix the ci issues

**Comment by rgplvr** [comment]
done

File: links/gql_dio_link/lib/src/dio_link.dart [hunk 1]
```diff
@@ -30,6 +31,7 @@ extension _CastDioResponse on dio.Response {
 class DioLink extends Link {
   /// Endpoint of the GraphQL service
   final String endpoint;
+  // final dio.CancelToken? cancelToken;
```
Comment (knaeckeKami) [inline links/gql_dio_link/lib/src/dio_link.dart:34]: pls delete commented leftover code

File: links/gql_dio_link/lib/src/dio_link.dart [hunk 2]
```diff
@@ -66,8 +68,7 @@ class DioLink extends Link {
 
   @override
   Stream<Response> request(Request request, [forward]) async* {
-    final dio.Response<Map<String, dynamic>> dioResponse =
-        await _executeDioRequest(
+    final dio.Response<Map<String, dynamic>> dioResponse = await _executeDioRequest(
```
Comment (knaeckeKami) [inline links/gql_dio_link/lib/src/dio_link.dart:71]: pls run `dart format`

**Review by knaeckeKami — CHANGES_REQUESTED** [review]
please delete the commented out code and run dart format

**Comment by rgplvr** [comment]
made the changes the source is properly formatted as per the diff

**Review by knaeckeKami — APPROVED** [review]

**Comment by knaeckeKami** [comment]
Thanks, LGTM

**Pull request merged**